### ISSUE TITLE
Stop reindexing indexes on new version

### ISSUE THREAD

**Issue by mhajas** [keycloak/keycloak#12497]
Currently, it is possible to configure Keycloak to reindex caches on startup. Reindexing basically means downtime because while reindexing is in progress, Keycloak is unusable. This can be replaced by a function introduced in Infinispan 14 - `udpateIndexSchema` this function is faster and makes no changes to index values, so there is no downtime.

**Comment by hmlnarik** [comment]
@mhajas Is this duplicate of #12067?

**Comment by mhajas** [comment]
Nope. Issue #12067 was about investigating how we will be doing no-downtime upgrades of indexes with HotRod. It resulted in scenarios we need to support and what steps we need to handle each scenario in no-downtime way. These scenarios will be added to ISPN testsuite so we are sure new versions don't break it for us. I will do a summary of the steps so we can use it when some changes appear.


Currently, we are not able to do no-downtime upgrades of indexes because we are not able to update the index schema (feature added in ISPN 14). Instead of it, with each startup, we reindex all caches just to make sure we have always the same schema. This reindexing may be time-consuming so we need to get rid of it and it will be done as part of this issue.

**Comment by hmlnarik** [comment]
Depends on #12494
